src/backend/distributed/operations/shard_split.c

```diff
@@ -701,10 +643,58 @@ BlockingShardSplit(SplitOperation splitOperation,
 }
 
 
+/* Check if a relation with given name already exists on the worker node */
+static bool
+CheckIfRelationWithSameNameExists(ShardInterval *shardInterval, WorkerNode *workerNode)
+{
+	char *schemaName = get_namespace_name(
+		get_rel_namespace(shardInterval->relationId));
+	char *shardName = get_rel_name(shardInterval->relationId);
+	AppendShardIdToName(&shardName, shardInterval->shardId);
+
+	StringInfo checkShardExistsQuery = makeStringInfo();
+	appendStringInfo(checkShardExistsQuery,
+					 "SELECT EXISTS (SELECT FROM pg_tables WHERE schemaname = '%s' AND tablename = '%s');",
+					 schemaName,
+					 shardName);
+
+	int connectionFlags = 0;
+	MultiConnection *connection = GetNodeUserDatabaseConnection(connectionFlags,
+																workerNode->workerName,
+																workerNode->workerPort,
+																CitusExtensionOwnerName(),
+																get_database_name(
+																	MyDatabaseId));
+
+	PGresult *result = NULL;
+	int queryResult = ExecuteOptionalRemoteCommand(connection,
+												   checkShardExistsQuery->data, &result);
+	if (queryResult != RESPONSE_OKAY || !IsResponseOK(result) || PQntuples(result) != 1)
+	{
+		ereport(ERROR, (errcode(ERRCODE_CONNECTION_FAILURE),
+						errmsg(
+							"Cannot check if relation %s already exists for split on worker %s:%d",
+							ConstructQualifiedShardName(shardInterval),
+							connection->hostname,
+							connection->port)));
+
+		PQclear(result);
+		ForgetResults(connection);
+
+		return false;
```

Comment:
This comes after an error, so it seems unlikely to be reached. ReportResultError handles this scenario somewhat gracefully.

Reply:
Fixed.